pkg/cmd/server/admin/create_kubeconfig.go

```diff
@@ -176,9 +177,15 @@ func (o CreateKubeConfigOptions) CreateKubeConfig() (*clientcmdapi.Config, error
 		ClientKeyData:         keyData,
 	}
 
+	// normalize the provided server to a format expected by config
+	serverNormalized, err := config.NormalizeServerURL(o.APIServerURL)
+	if err != nil {
+		return nil, err
+	}
+
 	clusters := make(map[string]*clientcmdapi.Cluster)
 	clusters[clusterNick] = &clientcmdapi.Cluster{
-		Server: o.APIServerURL,
+		Server: serverNormalized,
```

Comment:
A better approach would be to overwrite `o.APIServerURL` so that the code below sees the new value.